tools/check-style.sh

```diff
@@ -11,72 +11,60 @@
+if [ ! -z "$found" ]; then
+    # The mt=41 sets a red background for matched tabs:
+    echo -e '\033[31m\033[01mError: found CRLF characters in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo -e '\033[31m\033[01mError: found tabs instead of spaces in the following files:\033[0m'
+    done
+fi
 
-found=
-# The mt=41 sets a red background for matched trailing spaces
-GREP_COLORS='mt=41' GREP_COLOR='41' grep '[[:blank:]]\+$' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always |
-        while read f; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found trailing spaces in the following files:\033[0m'
-        found=1
-        errors=1
-    fi
 
-    echo "    $f"
-done
+found="$( grep -IUlr $'\r' include tests/*.{cpp,py,h} docs/*.rst --color=always )"
+if [ ! -z "$found" ]; then
+    echo -e '\033[31m\033[01mError: found CRLF characters in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
 
-found=
-grep '\<\(if\|for\|while\|catch\)(\|){' include tests/*.{cpp,py,h} -rn --color=always |
-        while read line; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: found the following coding style problems:\033[0m'
-        found=1
-        errors=1
-    fi
+found="$(GREP_COLORS='mt=41' GREP_COLOR='41' grep '[[:blank:]]\+$' include tests/*.{cpp,py,h} docs/*.rst -rn --color=always )"
+if [ ! -z "$found" ]; then
+    # The mt=41 sets a red background for matched trailing spaces
+    echo -e '\033[31m\033[01mError: found trailing spaces in the following files:\033[0m'
+    check_style_errors=1
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
 
-    echo "    $line"
-done
+found="$(grep '\<\(if\|for\|while\|catch\)(\|){' include tests/*.{cpp,py,h} -rn --color=always)"
+if [ ! -z "$found" ]; then
+    echo -e '\033[31m\033[01mError: found the following coding style problems:\033[0m'
+    check_style_errors=1
 
-found=
-GREP_COLORS='mt=41' GREP_COLOR='41' grep '^\s*{\s*$' include docs/*.rst -rn --color=always |
-        while read f; do
-    if [ -z "$found" ]; then
-        echo -e '\033[31m\033[01mError: braces should occur on the same line as the if/while/.. statement. Found issues in the following files: \033[0m'
-        found=1
-        errors=1
-    fi
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
 
-    echo "    $f"
-done
+found="$(GREP_COLORS='mt=41' GREP_COLOR='41' grep '^\s*{\s*$' include docs/*.rst -rn --color=always)"
+if [ ! -z "$found" ] ; then
+    check_style_errors=1
+    echo -e '\033[31m\033[01mError: braces should occur on the same line as the if/while/.. statement. Found issues in the following files: \033[0m'
+    echo "$found" | while read line; do
+        echo "    $line"
+    done
+fi
```

Comment:
This whole block is hitting false positives (both before and after the PR).  We aren't running it on `tests/*.{cpp,h}` basically because of this, and now, as you pointed out, we hit the same error in the documentation.

Adding a comment such as `{ // block` is really an ugly hack--it isn't a helpful comment, but is just serving to work around a defect in the style checker.

I think we're going to have to move beyond `grep` to fix this.  The following uses `awk` (with emulation of `grep`'s colour output) to do the check with a little more sophistication, but should get the behaviour we're after and lets us include the `cpp`/`h` files in `tests`:

```bash
found="$($(awk 'function prefix(filename, lineno) {
    return "    \033[35m" filename "\033[36m:\033[32m" lineno "\033[36m:\033[0m"
}
function mark(pattern, string) { sub(pattern, "\033[01;31m&\033[0m", string); return string }
last && /^\s*{/ {
    print prefix(FILENAME, FNR-1) mark("\\)\\s*$", last)
    print prefix(FILENAME, FNR)   mark("^\\s*{", $0)
    last=""
}
{ last = /(if|for|while|catch|switch)\s*\(.*\)\s*$/ ? $0 : "" }
' `find include -type f` tests/*.{cpp,h} docs/*.rst)"
if [ -n "$found" ]; then
    check_style_errors=1
    echo -e '\033[31;01mError: braces should occur on the same line as the if/while/.. statement. Found issues in the following files:\033[0m'
    echo "$found"
fi
```

Reply:
Fixed some typos, added.